burn-core/src/nn/sin_embedding.rs

```diff
@@ -0,0 +1,198 @@
+use alloc::vec::Vec;
+
+use crate as burn;
+use crate::config::Config;
+use crate::module::Module;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+use burn_tensor::Data;
+
+use libm::{cosf, expf, logf, sinf};
+
+/// Configuration to create an [SinEmbedding](SinEmbedding) layer.
+#[derive(Config)]
+pub struct SinEmbeddingConfig {
+    /// The number of embedding vectors.
+    n_embedding: usize,
```

Comment:
I would rename it to `max_sequence_size` since it's easier to know what value should be used. A default of 5000 is often common.